### ISSUE TITLE
Method signature types for methods of javax.persistence.AttributeConverter no longer included in configuration.

### ISSUE THREAD

**Issue by christophstrobl** [spring-projects-experimental/spring-native#1329]
spring-native: 0.11.0-SNAPSHOT

In the sample below, both `MonetaryAmountAttributeConverter` and `MonetaryAmount` are added to the configuration, but the more concrete return type `org.javamoney.moneta.Money` of  `convertToEntityAttribute(...)` ist not.
```java
@Converter(autoApply = true)
public class MonetaryAmountAttributeConverter implements AttributeConverter<MonetaryAmount, String> {

	@Override
	public String convertToDatabaseColumn(MonetaryAmount amount) {
		// ...
	}

	@Override
	public Money convertToEntityAttribute(String source) { // Money type not added to configuration
```

this leads to the following warn on startup
```
j.m.s.DefaultMonetaryAmountsSingletonSpi : Using Money as default amount type, because loading of configured org.javamoney.moneta.Money.defaults.amountType failed

java.lang.ClassNotFoundException: org.javamoney.moneta.Money
	at java.lang.Class.forName(DynamicHub.java:1433) ~[restbucks:na]
	at java.lang.Class.forName(DynamicHub.java:1408) ~[restbucks:na]
	at org.javamoney.moneta.spi.DefaultMonetaryAmountsSingletonSpi.loadDefaultAmountType(DefaultMonetaryAmountsSingletonSpi.java:66) ~[restbucks:1.4.2]
	at org.javamoney.moneta.spi.DefaultMonetaryAmountsSingletonSpi.<init>(DefaultMonetaryAmountsSingletonSpi.java:46) ~[restbucks:1.4.2]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[na:na]
```

**Comment by sdeleuze** [comment]
Could you a please share a minimal repro?

**Comment by christophstrobl** [comment]
closed via: 906925e by #1347
